src/methods.jl

```diff
@@ -77,3 +74,63 @@ julia> subtract_overscan(frame, (:, 4:5), dims = 2)
 * [`subtract_overscan!`](@ref)
 """
 subtract_overscan(frame::AbstractArray, idxs; dims = axes_min_length(idxs)) = subtract_overscan!(deepcopy(frame), idxs, dims = dims)
+
+
+"""
+    flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+
+In place version of [`flat_correct`](@ref)
+"""
+function flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+    size(frame) != size(flat_frame) && error("size of frame and flat frame are not same")
+    !(norm_value === nothing) && norm_value < 0 && error("norm_value must be greater than 0")
+
+    !(min_value === nothing) && @. flat_frame[flat_frame < min_value] = min_value
+
+    if !(norm_value === nothing)
+        flat_frame ./= norm_value
+    else
+        flat_frame ./= mean(flat_frame)
+    end
+
+    frame ./= flat_frame
+    return frame
+end
```

Comment:
On second look, I tihnk you can avoid having to parse the keyword arguments by doing this-
```julia
function flat_correct!(frame, flat_frame; norm_value=mean(flat_frame), min_value=nothing)
    if min_value !== nothing
       @. flat_frame[flat_frame < min_value] = min_value
    end
    flat_normed = flat_frame ./ norm_value
    ...
end
```

Reply:
Let me give this a try!

Reply:
I don't think this is a good idea, let me explain:
suppose `flat = ones(3,3)` and let `min_value = 2`, we will not be setting up the value of `norm_value` explicitly so it is by default `norm_value = 1`, whereas as `min_value !== nothing` the `flat_frame` gets modified and the mean would now change to `2` and the `norm_value` should have been equal to `2` whereas it is `1`. Hence we have an incoming trouble.

Reply:
Secondly, there is a fault with my version, it at times will act mutating for `flat_frame` as well (when some elements in the `flat_frame` is smaller than `min_value`), which I think is not something which a user would prefer. What are your thoughts on this? I was thinking of doing `_flat = deepcopy(flat_frame)` and then proceed in the usual way using `_flat`.

Reply:
This would be always mutating for `flat_frame` (except for a few cases) as we are dividing the `flat_frame` by `norm_value` (`flat_frame ./= norm_value`), should we go for a `deepcopy`?

Reply:
Resolved this issue by using fused broadcasts, i.e.
```
frame ./= (flat ./ norm_value)
```